packages/andromeda-data-storage/src/primitive.rs

```diff
@@ -48,6 +50,20 @@ pub enum Primitive {
     Binary(Binary),
 }
 
+impl Primitive {
+    pub fn from_string(&self) -> String {
```

Comment:
We should implement `From<Primitive>` for `String`, like what is done [here](https://github.com/andromedaprotocol/andromeda-core/blob/3ddabea8c77ca59981201a95a3748dab4b250911/packages/std/src/amp/addresses.rs#L285).

Reply:
Like this?
```rust
impl Primitive {
    pub fn from_string(&self) -> String {
        match self {
            Primitive::Uint128(_) => "Uint128".to_string(),
            Primitive::Decimal(_) => "Decimal".to_string(),
            Primitive::Coin(_) => "Coin".to_string(),
            Primitive::Addr(_) => "Addr".to_string(),
            Primitive::String(_) => "String".to_string(),
            Primitive::Bool(_) => "Bool".to_string(),
            Primitive::Binary(_) => "Binary".to_string(),
        }
    }
}

impl From<Primitive> for String {
    fn from(primitive: Primitive) -> Self {
        primitive.from_string()
    }
}

```

Reply:
Replace the `from_string` method completely:

```rust
impl From<Primitive> for String {
    fn from(primitive: Primitive) -> Self {
        match primitive{
            Primitive::Uint128(_) => "Uint128".to_string(),
            Primitive::Decimal(_) => "Decimal".to_string(),
            Primitive::Coin(_) => "Coin".to_string(),
            Primitive::Addr(_) => "Addr".to_string(),
            Primitive::String(_) => "String".to_string(),
            Primitive::Bool(_) => "Bool".to_string(),
            Primitive::Binary(_) => "Binary".to_string(),
        }
    }
}
```